docs/user/build-customization.rst

```diff
@@ -104,6 +105,60 @@ To avoid this, it's possible to unshallow the clone done by Read the Docs:
          - git fetch --unshallow
 
 
+Cancel build based on a condition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a command exits with code ``183``,
+Read the Docs will cancel the build immediately.
+You can use this approach to cancel builds that you don't want to complete based on some conditional logic.
+
+Here is an example that cancels builds from pull requests when there are no changes to the ``docs/`` folder compared to the ``origin/main`` branch:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.11"
+     jobs:
+       post_checkout:
+         # Cancel building pull requests when there aren't changed in the docs directory.
+         # `--quiet` exits with a 1 when there **are** changes,
+         # so we invert the logic with a !
+         #
+         # If there are no changes (exit 0) we force the command to return with 183.
+         # This is a special exit code on Read the Docs that will cancel the build immediately.
+         - |
+           if [ $READTHEDOCS_VERSION_TYPE = "external" ];
+           then
+             ! git diff --quiet origin/main -- docs/ && exit 183;
+           fi
+
+
+This other example shows how to cancel a build if the commit message contains ``skip ci`` on it:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.11"
+     jobs:
+       post_checkout:
+         # Use `git log` to check if the latest commit contains "skip ci",
+         # in that case exit the command with 183 to cancel the build
+         - |
+           case `git --no-pager log --pretty="tformat:%s" -1`
+           in *"skip ci"*)
+             exit 183;;
+           *);;
+           esac
```

Comment:
`case` has some rather gnarly syntax to it, is there any reason we're recommending this over `grep`?

```suggestion
           (git --no-pager log --pretty="tformat:%s" -1` | grep -q "skip ci") && exit 183
```

(Not tested against this PR, but should be close)

Reply:
You are the bash ninja. So, I'll let you decide what's better 👍

Reply:
Hah well I also haven't followed this part too closely, so you might have more reasoning for `case`. But if there is no strong preference, `grep` is more commonly used, so might be my suggestion.

My example was a little off, but this seems to be it:

```
(git --no-pager log --pretty="tformat:%s" -1 | grep -viq "skip ci") || exit 183
```

Note, the double inversion of `grep -v` and `||` to give either exit `0` on a miss or `183` on a hit of the search term:

```
(community) community/ main* 💀  git log -n 1                                                                                   
02dd7b34d (HEAD -> main, origin/main, origin/HEAD) Proxito: don't depend on attributes injected in the request (#9711) <stsewd@proton.me> 5 hours ago
(community) community/ main* 💀  /bin/sh -c '(git --no-pager log --pretty="tformat:%s" -1 | grep -viq "miss") || exit 183'
(community) community/ main* 💀  echo $?
0
(community) community/ main* 💀  /bin/sh -c '(git --no-pager log --pretty="tformat:%s" -1 | grep -viq "proxito") || exit 183'
(community) community/ main* 🔥  echo $?
183
```

Reply:
I don't think that grep example can do multiple directives? What if I want to match "skip ci" or "ci skip"?

Reply:
grep can do regex: `grep -E 'ci skip|skip ci'`

Reply:
I updated this example to use `grep`, 👍🏼

Reply:
Hmm. My shell-fu is weak and I cannot get this to work. It doesn't skip.

```
(git --no-pager log --pretty="tformat:%s -- %b" -1 | grep -vqE "skip rtd|rtd skip") || exit 183
```